File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.cs
using System;

public class Solution {
    public int solution(int[] array) {
        int answer = 0;
        int max = 1;
        int count = 1;
        int temp = -1;
        Array.Sort(array);
        
        if(array.Length == 1)
            return array[0];
        
        for(int i = 1; i < array.Length; i++)
        {
            if(array[i] == array[i - 1])
                count++;
            else
                count = 1;

            if(count == max)
                temp = -1;
            else if(count > max)
            {
                max = count;
                answer = array[i];
                temp = 0;
            }

        }
        return answer = temp == -1 ? -1 : answer;
    }
}

File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.cpp
#include <string>
#include <vector>
#include <queue>
#include <map>
using namespace std;

int solution(vector<int> array) {
    int answer = 0;
    if(array.size() == 1)
        return array.back();
    
    priority_queue<pair<int,int>> max_pq;
    map<int,int> map;
    
    for(auto i : array)
    {
        map[i] += 1;
        max_pq.push({map[i], i});
    }
    
    pair<int,int> pair1 = max_pq.top();
    max_pq.pop();
    pair<int,int> pair2 = max_pq.top();
    max_pq.pop();
    
    if(pair1.first == pair2.first)
        answer = -1;
    else
        answer = pair1.second;

    return answer;
}